pkg/catalog/service.go

```diff
@@ -100,16 +101,15 @@ func (mc *MeshCatalog) getServicesForServiceAccount(sa identity.K8sServiceAccoun
 // ListServiceIdentitiesForService lists the service identities associated with the given mesh service.
 func (mc *MeshCatalog) ListServiceIdentitiesForService(svc service.MeshService) ([]identity.ServiceIdentity, error) {
 	// Currently OSM uses kubernetes service accounts as service identities
-	serviceAccounts, err := mc.kubeController.ListServiceIdentitiesForService(svc)
-	if err != nil {
-		log.Err(err).Msgf("Error getting ServiceAccounts for Service %s", svc)
-		return nil, err
-	}
-
 	var serviceIdentities []identity.ServiceIdentity
-	for _, svcAccount := range serviceAccounts {
-		serviceIdentity := svcAccount.ToServiceIdentity()
-		serviceIdentities = append(serviceIdentities, serviceIdentity)
+	for _, provider := range mc.serviceProviders {
+		serviceIds, err := provider.ListServiceIdentitiesForService(svc)
+		if err != nil {
+			log.Err(err).Msgf("Error getting ServiceIdentities for Service %s", svc)
+			return nil, errServiceNotFound
```

Comment:
This error is not correct, the service identity error doesn't imply the service was not found.

Reply:
concretely, let's return the original error here